packages/samples/headless-react/src/components/query-summary/query-summary.class.tsx

```diff
@@ -33,20 +33,30 @@ export class QuerySummary extends Component {
 
   render() {
     const {
+      hasResults,
       hasQuery,
+      hasDuration,
       firstResult,
       lastResult,
       total,
       query,
       durationInSeconds,
     } = this.state;
-    return (
-      hasQuery && (
-        <p>
-          Results {firstResult}-{lastResult} of {total} for {query} in{' '}
-          {durationInSeconds} seconds
-        </p>
-      )
-    );
+
+    if (!hasResults) {
+      return null;
+    }
+
+    const summary = [`Results ${firstResult}-${lastResult} of ${total}`];
+
+    if (hasQuery) {
+      summary.push(`for ${query}`);
+    }
+
+    if (hasDuration) {
+      summary.push(`in ${durationInSeconds} seconds`);
+    }
+
+    return <p>{summary.join(' ')}</p>;
```

Comment:
What was the goal behind this change? I find the old version was much easier to read.

Reply:
When testing the tab, I noticed that `QuerySummary` wasn't showing any message due to a lack of query.

I decided to fix it and make sure it works in all possible situations: when there's no results, when there's results and a query, when there's results and a duration, when there's results and both a query & duration, when there's results and nothing else, etc.